tests/phpunit/tests/blocks/renderCommentTemplate.php

```diff
@@ -512,4 +512,110 @@ public function test_rendering_comment_template_unmoderated_preview() {
 			'Should not include any unapproved comments after removing filter'
 		);
 	}
+
+	public function test_rendering_comment_template_sets_comment_id_context() {
+		$parsed_comment_author_name_block = parse_blocks( '<!-- wp:comment-author-name /-->' )[0];
+		$comment_author_name_block        = new WP_Block(
+			$parsed_comment_author_name_block,
+			array(
+				'commentId' => self::$comment_ids[0],
+			)
+		);
+		$comment_author_name_block_markup = $comment_author_name_block->render();
+
+		$render_block_callback = static function( $block_content, $block ) use ( $parsed_comment_author_name_block ) {
+			/*
+			 * Insert a Comment Author Name block (which requires `commentId`
+			 * block context to work) after the Comment Content block.
+			 */
+			if ( 'core/comment-content' !== $block['blockName'] ) {
+				return $block_content;
+			}
+
+			$inserted_content = render_block( $parsed_comment_author_name_block );
+			return $inserted_content . $block_content;
+		};
+
+		add_filter( 'render_block', $render_block_callback, 10, 3 );
```

Comment:
No need to store the callback, just pass it directly here. The test suite cleans up all filters after each test, so no need to call `remove_filter()`

Reply:
Oh, you mean like inline? Cool, will do 👍

Reply:
756caa65ab